proposed/clock-meta.md

```diff
@@ -0,0 +1,154 @@
+time, that would allow interopability when testing code that relies on the current time 
+rather than relying installing PHP extensions or use hacks like re-declaring the `time()`
+function in other namespaces. 
+
+## 2. Why Bother?
+
+There are currently a few libraries that do provide the functionality on packagist, however 
+there is no interopability between these different libraries, as they ship with their own 
+clock interfaces. Symfony has a TimeMock library which uses namespace hacks to override the 
+`time()`, `date()`, `microtime()`, etc functions, however this does not solve mocking calls to 
+`new \DateTime()`
+
+Pros:
+
+* Consistent interface to get the current time
+* Easy to mock the wall clock time for repeatablility.
+
+Cons:
+
+* Extra overhead and developer effort to get the current time, not as simple as
+calling `time()` or `date()`.
+
+## 3. Scope
+
+### 3.1 Goals
+
+* Provide a simple and mockable way to read the current time
+* Allow interoperability between libraries when reading the clock
+
+### 3.2 Non-Goals
+
+* This PSR does not provide a recommendation on how and when to use the concepts
+  described in this document, so it is not a coding standard.
+* This PSR does not provide a recommendation on how to handle timezones when 
+  retrieving the current time. This is left up to the implementation.
+
+## 4. Approaches
+
+### 4.1 Chosen Approach
+
+We have decided to formalize the existing practices, use by several other packages
+out in the wild. Some of the popular packages providing this functionality are: 
+lcobucci/clock, kreait/clock, ergebnis/clock, and mangoweb/clock. Some providing
+interfaces, and some relying on overloading (extending) the Clock class to mock the
+current time.
+
+
+### 4.2 Example Implemntations
+
+```php
+final class TimeZoneAwareClock implements \Psr\Clock\ClockInterface
+{
+    private DateTimeZone $timeZone;
+
+    public function __construct(DateTimeZone $timeZone)
+    {
+        $this->timeZone = $timeZone;
+    }
+
+    public function now(): \DateTimeImmutable
+    {
+        return new DateTimeImmutable('now', $this->timeZone);
+    }
+}
+
+//
+
+final class SystemClock implements \Psr\Clock\ClockInterface
+{
+
+    public function now(): \DateTimeImmutable
+    {
+        return new \DateTimeImmutable();
+    }
+}
+
+//
+
+final class UTCClock implements \Psr\Clock\ClockInterface
+{
+    private TimeZoneAwareClock $inner;
+
+    public function __construct()
+    {
+        $this->inner = new TimeZoneAwareClock(new DateTimeZone('UTC'));
+    }
+
+    public function now(): \DateTimeImmutable
+    {
+        return $this->inner->now();
+    }
+}
+
+//
+
+final class FrozenClock implements \Psr\Clock\ClockInterface
+{
+    private DateTimeImmutable $now;
+    
+    public function __construct(DateTimeImmutable $now)
+    {
+        $this->now = $now;
+    }
+
+    public function now(): \DateTimeImmutable
+    {
+        return $this->now;
+    }
+}
+```
```

Comment:
```suggestion
\`\`\`php
use DateTimeImmutable;
use DateTimeZone;
use Psr\Clock\ClockInterface;

final class UTCClock implements ClockInterface
{
    public function now(): DateTimeImmutable
    {
        return new DateTimeImmutable('now', new DateTimeZone('UTC'));
    }
}

//

final class FrozenClock implements ClockInterface
{
    private DateTimeImmutable $now;
    
    public function __construct(DateTimeImmutable $now)
    {
        $this->now = $now;
    }

    public function now(): DateTimeImmutable
    {
        return $this->now;
    }
}
\`\`\`

While the `ClockInterface` does not strictly forbid to return `DateTimeImmutable` object with any timezone, we strongly recommend to use `UTC` as the default, you should refrain from either relying on default timezone which is dependent on OS settings or on a (e.g. do not use `new DateTimeImmutable()` without `new DateTimeZone('UTC')`).

And instead of having a clock with a dynamic timezone or multiple clock, simply rely on a common entry point and change the timezone on need:
\`\`\`php
$clock = new UTCClock();
$universalNow = $clock->now();
$nowForMyChicagoUser = $universalNow->setTimezone(new DateTimeZone('America/Chicago'));
\`\`\`

And the given `FrozenClock` can help you to mock "now":
\`\`\`php
$clock = new FrozenClock(new DateTimeImmutable('2039-01-02 10:00', new DateTimeZone('UTC')));
\`\`\`
```